uuid_v7.go

```diff
@@ -0,0 +1,182 @@
+package uuid
+
+import (
+	"bytes"
+	"crypto/rand"
+	"time"
+)
+
+type UUIDv7 uuidBase
+
+type UUIDv7Generator struct {
+	Precision int
+	currentTs []byte
+	counter   uint8
+}
```

Comment:
You should have a Mutex on here to protect that counter, which should probably be named `sequenceCounter`.

Precision should probably be something like SubsecondPrecisionBits - to make it perfectly clear what it is.  And you'll want to add the other bit lengths here - for sequence counter, node value, and random data.  And a way to set the static node value for the cases where that's needed (e.g. database with multiple nodes each with a number).  Put decent comments in as well, since this struct will be one of the more common places people will look for a reference on how different fields are dealt with.